addons/website_profile/controllers/main.py

```diff
@@ -168,23 +168,27 @@ def _prepare_badges_domain(self, **kwargs):
             domain = expression.AND([[('challenge_ids.category', '=', kwargs.get('badge_category'))], domain])
         return domain
 
-    @http.route('/profile/ranks_badges', type='http', auth="public", website=True)
-    def view_ranks_badges(self, **kwargs):
+    def _prepare_ranks_badges_values(self, **kwargs):
         ranks = []
         if 'badge_category' not in kwargs:
             Rank = request.env['gamification.karma.rank']
             ranks = Rank.sudo().search([], order='karma_min DESC')
 
         Badge = request.env['gamification.badge']
         badges = Badge.sudo().search(self._prepare_badges_domain(**kwargs))
-        badges = sorted(badges, key=lambda b: b.stat_count_distinct, reverse=True)
+        badges.sorted("granted_users_count", reverse=True)
```

Comment:
I think `.sorted` only returns the sorted value, without altering the initial browserecord.
So it should be:

`badges = badges.sorted("granted_users_count", reverse=True)`